runtime/drivers/s3/model_manager.go

```diff
@@ -0,0 +1,113 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelResultProperties struct {
+	Path string `mapstructure:"path"`
+}
+
+func (c *Connection) Rename(ctx context.Context, res *drivers.ModelResult, newName string, env *drivers.ModelEnv) (*drivers.ModelResult, error) {
+	return nil, nil
+}
+
+func (c *Connection) Exists(ctx context.Context, res *drivers.ModelResult) (bool, error) {
+	return true, nil
+}
+
+func (c *Connection) Delete(ctx context.Context, res *drivers.ModelResult) error {
+	cfg, err := c.awsConfig(ctx, "")
+	if err != nil {
+		return err
+	}
+	p := &ModelResultProperties{}
+	if err := mapstructure.Decode(res.Properties, p); err != nil {
+		return err
+	}
+	u, err := url.Parse(p.Path)
+	if err != nil {
+		return err
+	}
+	base, _ := doublestar.SplitPattern(strings.TrimPrefix(u.Path, "/"))
+	return deleteObjectsInPrefix(ctx, cfg, u.Host, base)
+}
+
+func deleteObjectsInPrefix(ctx context.Context, cfg aws.Config, bucketName, prefix string) error {
+	s3client := s3.NewFromConfig(cfg)
+
+	deleteBatch := func(objects []types.ObjectIdentifier) error {
+		_, err := s3client.DeleteObjects(ctx, &s3.DeleteObjectsInput{
+			Bucket: &bucketName,
+			Delete: &types.Delete{
+				Objects: objects,
+			},
+		})
+		return err
+	}
+
+	var continuationToken *string
+	for {
+		out, err := s3client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+			Bucket:            &bucketName,
+			Prefix:            &prefix,
+			ContinuationToken: continuationToken,
+		})
+		if err != nil {
+			return err
+		}
+
+		ids := make([]types.ObjectIdentifier, 0, len(out.Contents))
+		for _, o := range out.Contents {
+			ids = append(ids, types.ObjectIdentifier{
+				Key: o.Key,
+			})
+		}
+
+		if len(ids) > 0 {
+			if err := deleteBatch(ids); err != nil {
+				return err
+			}
+		}
+
+		if *out.IsTruncated && out.NextContinuationToken != nil {
+			continuationToken = out.NextContinuationToken
+		} else {
+			break
+		}
+	}
+
+	return nil
+}
+
+func (c *Connection) awsConfig(ctx context.Context, awsRegion string) (aws.Config, error) {
```

Comment:
Can it use `getAwsSessionConfig(...).Config` instead so we don't have duplicate functions for initializing AWS configs? (Note the func is renamed to `newSessionForBucket` in my glob PR.)

Reply:
Hmm. I copied it from athena and planned to refactor all of these into `s3` pkg with the athena/redshift connector changes.